### PR TITLE
Fix: rework pre chat activity layout to rely on author entity

### PR DESCRIPTION
**Pull request by OEvgeny**
### Changelog entry

- Reworked pre-chat activity layout to use author entity for improved consistency and flexibility, in PR [#5274](https://github.com/microsoft/BotFramework-WebChat/pull/5274), by [@OEvgeny](https://github.com/OEvgeny)

## Description

This PR enhances the pre-chat activity layout by leveraging the author entity along with some other minor fixes and improvements

## Design

The redesign focuses on using the `author` property within the `CreativeWork` entity to populate pre-chat message details along the following:

1. Displaying the bot's avatar, name, and description
2. Improved styling for placeholder states
3. Enhanced accessibility for liner activities

## Specific Changes

- Updated `PreChatMessageActivity` component to use author information for layout
- Modified `CreativeWork` type to support `Person` as an author type
- Added transition to card actions
- Adjusted CSS styles for pre-chat message activities, including avatar, title, and subtitle
- Added styling for liner activity focus states
- Updated test files to include author information in pre-chat messages
- Refined `StarterPromptsCardAction` component styling

## -

- [x] I have added tests and executed them locally
- [x] I have updated `CHANGELOG.md`
- [ ] ~I have updated documentation~

### CODE REVIEW

File: packages/core/src/types/external/OrgSchema/CreativeWork.ts [hunk 1]
```diff
@@ -71,13 +71,29 @@ export type CreativeWork = Thing & {
...
+  name?: string | undefined;
+};
+
+const person = <TEntries extends ObjectEntries>(entries?: TEntries | undefined) =>
+  object({
+    description: orgSchemaProperty(string()),
+    image: orgSchemaProperty(string()),
+    name: orgSchemaProperty(string()),
+
+    ...entries
```
Comment (compulim) [inline packages/core/src/types/external/OrgSchema/CreativeWork.ts:87]: Maybe we don't need this right now.

File: packages/core/src/types/external/OrgSchema/CreativeWork.ts [hunk 2]
```diff
@@ -71,13 +71,29 @@ export type CreativeWork = Thing & {
...
+  });
+
 export const creativeWork = <TEntries extends ObjectEntries>(entries?: TEntries | undefined) =>
   thing({
     // For forward compatibility, we did not enforce @type must be "CreativeWork" or any other subtypes.
     // In future, if Schema.org introduced a new subtype of CreativeWork, we should still able to parse that one as a CreativeWork.
 
     abstract: orgSchemaProperty(string()),
-    author: orgSchemaProperties(string()),
+    author: orgSchemaProperty(union([lazy(() => person()), string()])),
```
Comment (compulim) [inline packages/core/src/types/external/OrgSchema/CreativeWork.ts:96]: Maybe we don't need `lazy()` as we don't have recursive yet.

File: packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx [hunk 3]
```diff
@@ -10,29 +10,40 @@ import StarterPromptsCardAction from './StarterPromptsCardAction.js';
 
 type Props = Readonly<{ activity: WebChatActivity & { type: 'message' } }>;
 
-const { useRenderMarkdownAsHTML } = hooks;
+const { useRenderMarkdownAsHTML, useLocalizer } = hooks;
```
Comment (compulim) [inline packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx:13]:
```suggestion
const { useLocalizer, useRenderMarkdownAsHTML } = hooks;
```

File: packages/core/src/types/external/OrgSchema/CreativeWork.ts (hunk 2 above)
Comment (OEvgeny) [reply packages/core/src/types/external/OrgSchema/CreativeWork.ts:96]:
```suggestion
    author: orgSchemaProperty(union([person(), string()])),
```

File: packages/core/src/types/external/OrgSchema/CreativeWork.ts (hunk 1 above)
Comment (OEvgeny) [reply packages/core/src/types/external/OrgSchema/CreativeWork.ts:87]: Let's keep it for consistency

File: packages/fluent-theme/src/components/preChatActivity/StarterPromptsCardAction.module.css [hunk 4]
```diff
@@ -15,6 +15,7 @@
   padding: 16px 20px;
   text-align: left;
   user-select: none;
+  transition: background-color var(--webchat-durationNormal) var(--webchat-curveDecelerateMid);
```
Comment (compulim) [inline packages/fluent-theme/src/components/preChatActivity/StarterPromptsCardAction.module.css:18]:
```suggestion
  text-align: left;
  transition: background-color var(--webchat-durationNormal) var(--webchat-curveDecelerateMid);
  user-select: none;
```

File: packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx [hunk 5]
```diff
@@ -10,29 +10,40 @@ import StarterPromptsCardAction from './StarterPromptsCardAction.js';
...
-        dangerouslySetInnerHTML={html}
-      />
+      {author && (
+        <div
+          className={cx(
+            classNames['pre-chat-message-activity__body'],
+            isPlaceHolder && classNames['pre-chat-message-activity__body--placeholder']
+          )}
+        >
+          <img
```
Comment (compulim) [inline packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx:38]: <3

File: packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx [hunk 6]
```diff
@@ -10,29 +10,40 @@ import StarterPromptsCardAction from './StarterPromptsCardAction.js';
...
+            alt={localize('AVATAR_ALT', author.name)}
+            className={classNames['pre-chat-message-activity__body-avatar']}
+            src={author.image}
+          />
+          <h2 className={classNames['pre-chat-message-activity__body-title']}>{author.name}</h2>
```
Comment (compulim) [inline packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx:43]: Should we hide `<h2>` if `author.name` is falsy?

File: packages/core/src/types/external/OrgSchema/CreativeWork.ts [hunk 7]
```diff
@@ -26,7 +26,7 @@ export type CreativeWork = Thing & {
    *
    * @see https://schema.org/author
    */
-  author?: string | undefined;
+  author?: string | Person | undefined;
```
Comment (compulim) [inline packages/core/src/types/external/OrgSchema/CreativeWork.ts:29]:
```suggestion
  author?: Person | string | undefined;
```

File: packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx [hunk 8]
```diff
@@ -10,29 +10,40 @@ import StarterPromptsCardAction from './StarterPromptsCardAction.js';
...
-
-  const html = useMemo(
-    () => (renderMarkdownAsHTML ? { __html: renderMarkdownAsHTML(activity.text || '') } : { __html: '' }),
-    [activity.text, renderMarkdownAsHTML]
-  );
+  const localize = useLocalizer();
 
   const entity = getOrgSchemaMessage(activity?.entities || []);
   const isPlaceHolder = entity?.keywords?.includes('PreChatMessage') && entity.creativeWorkStatus === 'Placeholder';
+  const author = typeof entity?.author !== 'string' ? entity?.author : undefined;
```
Comment (compulim) [inline packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx:22]: Maybe if `typeof author === 'string'`, we treat it as `{ name: author, type: 'Person' }` then call it done?

**Review by compulim — DISMISSED** [review]
Nitpicking

File: packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx (hunk 8 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/preChatActivity/PreChatMessageActivity.tsx:22]: Good idea. Making changes

File: packages/fluent-theme/src/components/preChatActivity/StarterPromptsCardAction.module.css [hunk 9]
```diff
@@ -14,6 +14,7 @@
   overflow: hidden;
   padding: 16px 20px;
   text-align: left;
+  transition: background-color var(--webchat-durationNormal) var(--webchat-curveDecelerateMid);
```
Comment (compulim) [inline packages/fluent-theme/src/components/preChatActivity/StarterPromptsCardAction.module.css:17]: ❤️

**Review by compulim — APPROVED** [review]

**Pull request merged**